benchmarks/Microsoft.AspNetCore.Routing.Performance/Matchers/JumpTableMultipleEntryBenchmark.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using BenchmarkDotNet.Attributes;
+
+namespace Microsoft.AspNetCore.Routing.Matchers
+{
+    public class JumpTableMultipleEntryBenchmark
+    {
+        private string[] _strings;
+        private PathSegment[] _segments;
+        private int[] _samples;
+
+        private JumpTable _linearSearch;
+        private JumpTable _dictionary;
+
+        // All factors of 100 to support sampling
+        [Params(2, 5, 10, 25, 50, 100)]
+        public int Count;
```

Comment:
FYI the breakpoint for dictionary == linear search is 10. This is high because of the substring

I'm not including any of the more esoteric implementations in this PR because those will require really detailed tests and review. Everything here is pretty simple.